src/content-service/services/__tests__/repository-provider.spec.ts

```diff
@@ -84,4 +91,36 @@ describe('Content repository', () => {
       expect(Product).toBeCalledWith(product);
     });
   });
+
+  describe('getProductFromV3 function', () => {
+    it('should throw an error if product id is not defined', async () => {
+      const repository = getContentServiceRepository(Environment.Dev);
+      expect.assertions(1);
+      await expect(repository.getProductFromV3(null)).rejects.toEqual(new Error('getProduct: product id is required'));
+    });
+
+    it('should return null if product v3 was not found', async () => {
+      (getContentServiceApi as jest.Mock).mockImplementationOnce(() => ({ getProductFromV3: jest.fn(() => null) }));
+      const repository = getContentServiceRepository(Environment.Dev);
+      const result = await repository.getProductFromV3(id);
+
+      expect(result).toBeNull();
+    });
+
+    it('should return product v3', async () => {
+      const repository = getContentServiceRepository(Environment.Dev);
+      await repository.getProductFromV3(productV3.id)
+
+      expect(getProductFromV3).toBeCalledWith(productV3.id);
+      expect(ProductV3).toBeCalledWith(productV3);
+    });
+
+    it('should return product v3', async () => {
```

Comment:
Am I crazy or these two lasts tests are exactly equal? 🤔

Reply:
you are not :). Fixed